src/fieldset.c

```diff
@@ -206,6 +206,16 @@ void fs_chkadd_string(fieldset_t *fs, const char *name, char *value, int free_)
 	}
 }
 
+void fs_chkadd_unsafe_string(fieldset_t *fs, const char *name, char *value, int free_)
+{
+	if (value) {
+		fs_add_unsafe_string(fs, name, value, free_);
```

Comment:
To clarify: I mean empty strings do not result in an `fs_add_null`, which I'm not sure if is intended behavior or not.